src/index.test.ts

```diff
@@ -101,12 +107,32 @@ describe('Stripe module loader', () => {
       const {loadStripe} = require('./index');
       const stripePromise = loadStripe('pk_test_foo');
 
-      return Promise.resolve().then(() => {
+      return Promise.resolve().then(async () => {
         dispatchScriptEvent('error');
 
-        return expect(stripePromise).rejects.toEqual(
+        await expect(stripePromise).rejects.toEqual(
+          new Error('Failed to load Stripe.js')
+        );
+        expect(console.warn).not.toHaveBeenCalled();
+      });
+    });
+
+    it('does not cause unhandled rejects when the script fails', () => {
+      const consoleWarnCalled = new Promise((resolve) => {
+        consoleWarnSpy.mockImplementation(resolve);
```

Comment:
Huh, why isn't the `jest.spyOn(console, 'warn')` in the `beforeEach` block sufficient to determine if `console.warn` was called here?

Reply:
`console.warn` is called asynchronously when the promise inside `stripePromise` resolves. The other test awaits the `loadStripe` rejection, so it's ensured the internal `catch` of `stripePromise` that triggers `console.warn` would have been called by then.